ql/src/java/org/apache/hadoop/hive/ql/parse/ImportSemanticAnalyzer.java

```diff
@@ -772,10 +772,21 @@ we use isExternal and not tableType() method since that always gives type as man
         tgtPath = new Path(table.getDataLocation().toString(),
             Warehouse.makePartPath(partSpec.getPartSpec()));
       } else {
-        Database parentDb = x.getHive().getDatabase(tblDesc.getDatabaseName());
-        tgtPath = new Path(
-            wh.getDefaultTablePath( parentDb, tblDesc.getTableName(), tblDesc.isExternal()),
-            Warehouse.makePartPath(partSpec.getPartSpec()));
+        try {
+          Table translatedTable =
```

Comment:
can we extract the get location logic and the construct the. final tgt path?
````
String path = getLocation(tblDesc, partSpec, table);
tgtPath = new Path(path, Warehouse.makePartPath(partSpec.getPartSpec()));
````

Reply:
sure will do that